perfkitbenchmarker/linux_virtual_machine.py

```diff
@@ -954,3 +956,191 @@ def MoveFile(self, target, source_path, remote_path=''):
 
     # Copies the file to its final destination in the container
     target.ContainerCopy(file_name, remote_path)
+
+
+class JujuMixin(DebianMixin):
+    """
+    Class to allow running juju-deployed workloads
+
+    Bootstraps a Juju environment using the manual provider:
+    https://jujucharms.com/docs/stable/config-manual
+    """
+
+    OS_TYPE = 'juju'
+
+    isController = False
+
+    """
+    A reference to the juju controller, useful when operations occur against
+    a unit's VM but need to be preformed from the controller.
+    """
+    controller = None
+
+    units = []
+
+    environments_yaml = """
+    default: perfkit
+
+    environments:
+        perfkit:
+            type: manual
+            bootstrap-host: {0}
+    """
+
+    def juju_bootstrap(self):
```

Comment:
Nit: to keep consistent with the rest of the codebase could you please use CamelCase for methods? And if the method is private, you can prefix the method name with '_'.